### DISCUSSION TITLE
Image Field Preview: Cors required for edit but not detail

### DISCUSSION THREAD

**Discussion by stuartcusackie** [laravel/nova-issues#5301] [Q&A] [answered]
- Laravel Version:  9.50.2
- Nova Version: 4.21.0
- PHP Version: 8.1.12

### Description:

This is not a major issue at all and easy to fixed by adding `GET *` to my DO Spaces CORS configuration. It seems like a bug but I might be wrong.

Image field previews are not showing in Resource Edit view when using DO Spaces -  unless you set the CORS config as described above. I find this weird because image field previews work perfectly fine in Resource Detail view.

**Screenshot 1 (Detail mode - preview image visible):**
![image](https://user-images.githubusercontent.com/20210934/217397377-6f9c2b24-6525-4cb7-86f6-1e6a5c713bed.png)

**Screenshot 2 (Edit mode - preview image not visible - It should be above the blue button):**
![image](https://user-images.githubusercontent.com/20210934/217397411-37d19794-75ad-48ac-bd23-63c3a841c7a2.png)

I can see CORS errors when I go to edit mode:
![image](https://user-images.githubusercontent.com/20210934/217398328-7ea4eac0-3ed5-44b1-9242-c3ba23e710d4.png)
but everything is fine in detail mode. Weird because they fetch the exact same url!

My Image field configuration is unusual because it integrates with Spatie's Media Libary, but I don't think this is the problem. As mentioned, everything works when using the local disk, and detail mode works fine with remote disks.

## Possible cause

I guess the detail view is fetching the image in a different way to the edit view. I checked Chrome's network tab to compare the requests. The edit mode is adding the domain name to the origin, whereas detail mode has no origin:

Edit Mode ('origin' causing the problem?):
![image](https://user-images.githubusercontent.com/20210934/217398989-3121e0d2-71ae-4fe2-aba7-d22022fb91d1.png)

Preivew Mode (No origin - preview image works!)
![image](https://user-images.githubusercontent.com/20210934/217399105-87d8c1f7-22cd-4e7d-9721-3b094058655c.png)

**Comment by crynobone** [comment, marked as answer]
Hi there,

The detail and edit page doesn't use the same Vue component code to preview the image which introduces the differences here. In order to support previewing stored images and previewing selected images from the local disk we are using `URL.createObjectURL()` which could be the cause of requiring `origin`.

I believe setting the correct CORS for cloud storage is the way to go here.

**Reply by stuartcusackie** [reply]
That makes perfect sense. Thanks for the explanation.
